frontend/src/store/modules/shoots/index.js

```diff
@@ -324,15 +324,32 @@ function setFilteredItems (state, rootState, rootGetters) {
   state.filteredShoots = items
 }
 
-const putItem = (state, newItem) => {
-  const item = findItem(state)(newItem.metadata)
-  if (item !== undefined) {
-    if (item.metadata.resourceVersion !== newItem.metadata.resourceVersion) {
-      Vue.set(state.shoots, keyForShoot(item.metadata), assign(item, newItem))
+const putItem = (state, item) => {
+  const shoot = findItem(state)(item.metadata)
+  const key = keyForShoot(item.metadata)
+  if (shoot !== undefined) {
+    if (shoot.metadata.resourceVersion !== item.metadata.resourceVersion) {
+      Vue.set(state.shoots, key, assign(shoot, item))
     }
   } else {
+    item.info = undefined // register property to ensure reactivity
+    Vue.set(state.shoots, key, item)
+  }
+}
```

Comment:
We should not mix a renaming which has nothing to do with the performance improvement in the same commit. It is very confusing when reviewing the changes.